pennylane/ops/op_math/composite.py

```diff
@@ -255,3 +255,8 @@ def hash(self):
     @property
     def arithmetic_depth(self) -> int:
         return 1 + max(op.arithmetic_depth for op in self)
+
+    def map_wires(self, wire_map: dict):
+        new_op = super().map_wires(wire_map=wire_map)
+        new_op.operands = tuple(op.map_wires(wire_map=wire_map) for op in self)
```

Comment:
This seems like it's doing a lot more work than we need to do.  `super().map_wires` is going to call copy, which is going to copy all the operations.  And then we immediately get rid of those operands that we just created.  

I think it might be worth just replicating some of the code from `__copy__` here so we don't have to duplicate the operands twice.